### ISSUE TITLE
Change errors to error chain

### ISSUE THREAD

**Issue by brandur** [brandur/redis-cell#20]
Move the error system over to [error chain](https://github.com/rust-lang-nursery/error-chain) as it's starting to become a de facto standard in Rust and provides better introspection and some nice ergonomic improvements for projects using it.

**Comment by gsquire** [comment]
Another good candidate could be [failure](https://github.com/withoutboats/failure) just FYI.

**Comment by brandur** [comment]
> Another good candidate could be failure just FYI.

@gsquire Yep, good point. I don't have a strong preference, but if at all possible would like it'll be longest-lived and best-maintained. That might be either of them, but just given that it's a relatively new project, I might give `failure` a little more time to make sure that it makes it through its phase of adolescence.
